Sources/BraintreeCore/BTHTTP.swift

```
@@ -151,30 +159,34 @@ class BTHTTP: NSObject, URLSessionTaskDelegate {
         path: String,
         configuration: BTConfiguration? = nil,
         parameters: [String: Any]? = [:],
-        headers: [String: String]? = nil,
-        completion: RequestCompletion?
-    ) {
-        do {
-            let request = try createRequest(
-                method: method,
-                path: path,
-                configuration: configuration,
-                parameters: parameters,
-                headers: headers
-            )
+        headers: [String: String]? = nil
+    ) async throws -> (BTJSON?, HTTPURLResponse?) {
```

Are we able to make `BTJSON` and `HTTPURLResponse` return types non null? They might have been optional before since we could return an Error in the completion handler.

---

Gotcha, I'll make the update